sklearn/gaussian_process/_gpr.py

```diff
@@ -110,6 +110,14 @@ def optimizer(obj_func, initial_theta, bounds):
         which might cause predictions to change if the data is modified
         externally.
 
+    n_targets : int, default=None
+        The number of dimensions of the target values. Used to decide the number
+        of outputs when sampling from the prior distributions (i.e. calling
+        :meth:`sample_y` before :meth:`fit`). This parameter is ignored once
+        :meth:`fit` has been called.
```

Comment:
API-wise, ignoring `n_targets` after `fit` is strange to me.

@glemaitre Assuming `n_targets` is set, what do you think of requiring the number of targets seen in `fit` to be consistent with `n_targets`?

Reply:
This is less lenient and more conservative but I don't have any objection to that.
@MaxwellLZH Do you want to make this change of behaviour?

Reply:
I'm also in favor of checking consistency, this is more in line with the usual behavior of scikit-learn estimators.